Recidiviz/justice-counts
# [#771][Agency Dashboard] Only allowlisted agencies should have a dashboard (all others 404)

## Description of the change

Adds corrections and feedback from previous PR 

Closes #771

## Test Plan

Test at https://alfonsotest---agency-dashboard-web-b47yvyxs3q-uc.a.run.app/

I added staging to the allowlist logic as well so this is testable before merging. 

- Navigate to https://alfonsotest---agency-dashboard-web-b47yvyxs3q-uc.a.run.app/agency/department-of-corrections-%5Bcla-demo%5D/dashboard
- Verify that it loads.
- Navigate to a different agency not on the allowlist. 
- Verify that it 404s

## Checklists

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [x] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:
